azext_iot/iothub/providers/state.py

```diff
@@ -173,34 +239,50 @@ def upload_hub_from_dict(self, hub_state: dict):
                 module_identity = modules[j][0]
                 module_twin = modules[j][1]
 
-                self.upload_module_identity(module_identity)
+                self.upload_module_identity(identity["deviceId"], module_identity)
 
                 _iot_device_module_twin_replace(target=self.target, device_id=identity["deviceId"],
                                                 module_id=module_identity["module_id"], target_json=json.dumps(module_twin))
 
         # set parent-child relationships
-        for parentId in hub_state["children"]:
-            child_list = hub_state["children"][parentId]
+        for parentId in hub_state["devices"]["children"]:
+            child_list = hub_state["devices"]["children"][parentId]
             _iot_device_children_add(target=self.target, device_id=parentId, child_list=child_list)
 
-    def save_state(self, filename: str, overwrite_file=False):
+        # CONTROL PLANE
```

Comment:
When we have the control plane sdk in the extension, this will drastically change (as we may be able to do more steps at once rather than one by one)